src/core/etl/tests/Flow/ETL/Tests/Unit/RowsTest.php

```diff
@@ -285,6 +285,65 @@ public function test_building_rows_from_array_with_schema_but_entries_not_availa
         );
     }
 
+    public function test_building_single_row_from_array() : void
+    {
+        $row = array_to_row(
+            [
+                ['id' => 1234, 'deleted' => false, 'phase' => null],
+                ['id' => 4321, 'deleted' => true, 'phase' => 'launch'],
+            ]
```

Comment:
hmm I'm honestly not sure how this edge case should be handled, because technically speaking this is a use case for `array_to_rows()` and the output that returns only one row when input defines two of them is a bit hard to predict. 

I'm tempted to say that in this case array_to_row should throw an exception. 

The validation could simply check if all array keys are strings (since only strings can become a entry names)